rmw_connext_shared_cpp/test/test_security_logging.cpp

```diff
@@ -0,0 +1,212 @@
+      "INVALID_VERBOSITY is not a supported verbosity"));
+}
+
+TEST_F(SecurityLoggingTest, test_log_distribute)
+{
+  std::string xml_file_path = write_logging_xml("<distribute>true</distribute>");
+  DDS::PropertyQosPolicy policy;
+  EXPECT_EQ(apply_logging_configuration_from_file(xml_file_path.c_str(), policy), RMW_RET_OK);
+  EXPECT_FALSE(rmw_error_is_set());
+
+  EXPECT_EQ(log_file_property(policy), nullptr);
+  EXPECT_EQ(verbosity_property(policy), nullptr);
+  EXPECT_STREQ(logging_distribute_enable_property(policy), "true");
+  EXPECT_EQ(logging_distribute_depth_property(policy), nullptr);
+}
+
+TEST_F(SecurityLoggingTest, test_log_depth)
+{
+  std::string xml_file_path = write_logging_xml("<qos><depth>10</depth></qos>");
+  DDS::PropertyQosPolicy policy;
+  EXPECT_EQ(apply_logging_configuration_from_file(xml_file_path.c_str(), policy), RMW_RET_OK);
+  EXPECT_FALSE(rmw_error_is_set());
+
+  EXPECT_EQ(log_file_property(policy), nullptr);
+  EXPECT_EQ(verbosity_property(policy), nullptr);
+  EXPECT_EQ(logging_distribute_enable_property(policy), nullptr);
+  EXPECT_STREQ(logging_distribute_depth_property(policy), "10");
+}
+
+TEST_F(SecurityLoggingTest, test_profile)
+{
+  std::string xml_file_path = write_logging_xml("<qos><profile>DEFAULT</profile></qos>");
+  DDS::PropertyQosPolicy policy;
+  EXPECT_EQ(apply_logging_configuration_from_file(xml_file_path.c_str(), policy), RMW_RET_OK);
+  EXPECT_FALSE(rmw_error_is_set());
+
+  EXPECT_EQ(log_file_property(policy), nullptr);
+  EXPECT_EQ(verbosity_property(policy), nullptr);
+  EXPECT_EQ(logging_distribute_enable_property(policy), nullptr);
+  EXPECT_STREQ(logging_distribute_depth_property(policy), "10");
+}
+
+TEST_F(SecurityLoggingTest, test_profile_overwrite)
+{
+  std::string xml_file_path = write_logging_xml(
+    "<qos>\n"
+    "  <profile>DEFAULT</profile>\n"
+    "  <depth>42</depth>\n"
+    "</qos>");
+  DDS::PropertyQosPolicy policy;
+  EXPECT_EQ(apply_logging_configuration_from_file(xml_file_path.c_str(), policy), RMW_RET_OK);
+  EXPECT_FALSE(rmw_error_is_set());
+
+  EXPECT_EQ(log_file_property(policy), nullptr);
+  EXPECT_EQ(verbosity_property(policy), nullptr);
+  EXPECT_EQ(logging_distribute_enable_property(policy), nullptr);
+  EXPECT_STREQ(logging_distribute_depth_property(policy), "42");
+}
+
+TEST_F(SecurityLoggingTest, test_profile_invalid)
+{
+  std::string xml_file_path = write_logging_xml("<qos><profile>INVALID_PROFILE</profile></qos>");
+  DDS::PropertyQosPolicy policy;
+  EXPECT_EQ(apply_logging_configuration_from_file(xml_file_path.c_str(), policy), RMW_RET_ERROR);
+  EXPECT_TRUE(rmw_error_is_set());
+  EXPECT_THAT(rmw_get_error_string().str, HasSubstr("INVALID_PROFILE is not a supported profile"));
+}
+
+TEST_F(SecurityLoggingTest, test_all)
+{
+  std::string xml_file_path = write_logging_xml(
+    "<file>foo</file>\n"
+    "<verbosity>ERROR</verbosity>\n"
+    "<distribute>true</distribute>\n"
+    "<qos>\n"
+    "  <depth>10</depth>\n"
+    "</qos>");
```

Comment:
@mikaelarguedas this is using the original format. I'll switch to the nested one as soon as I get some clarification in the design doc.

Reply:
This is now done.